x/capability/keeper/keeper.go

```diff
@@ -117,12 +117,22 @@ func (k *Keeper) InitializeAndSeal(ctx sdk.Context) {
 	k.sealed = true
 }
 
-// SetIndex sets the index to one in InitChain
-// Since it is an exported function, we check that index is indeed unset, before initializing
-func (k Keeper) SetIndex(ctx sdk.Context, index uint64) {
+// SetIndex sets the index to one (or greater) in InitChain according
+// to the GenesisState. It must only be called once.
+// It will panic if the provided index is 0, or if the index is already set.
+func (k Keeper) SetIndex(ctx sdk.Context, index uint64) error {
+	if index == 0 {
+		panic("SetIndex requires index > 0")
+	}
+	latest := k.GetLatestIndex(ctx)
+	if latest > 0 {
```

Comment:
nit

```suggestion

	if latest := k.GetLatestIndex(ctx); latest > 0 {
```

Reply:
There isn't even a need to store this in a variable.

Reply:
I kind of prefer the clarity of how its written currently for symmetry with the other `index` value being evaluated.